### PR TITLE
[INLONG-6657][Sort] Add dirty data metric for Hive

### LINKED ISSUE #6657: [Improve][Sort] Add dirty data metric for Hive
### Description

Add dirty data metric for Hive.

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haifxu**
### Prepare a Pull Request

- Fixes #6657 

### Motivation

Add dirty data metric for Hive.

### Modifications

1. Add `DirtyRecords` and `DirtyBytes` when init `MetricOption`.
2. Calculate dirty data when an exception is caught.

### CODE REVIEW

File: inlong-sort/sort-connectors/hive/src/main/java/org/apache/inlong/sort/hive/filesystem/AbstractStreamingWriter.java [hunk 1]
```diff
@@ -195,14 +199,21 @@ public void processWatermark(Watermark mark) throws Exception {
...
+                    element.hasTimestamp() ? element.getTimestamp() : null,
+                    currentWatermark);
+            if (metricData != null) {
+                metricData.invokeWithEstimate(element.getValue());
+            }
+        } catch (Exception e) {
+            if (metricData != null) {
+                metricData.invokeDirtyWithEstimate(element.getValue());
+            }
+            throw new RuntimeException(e);
```
Comment (gong) [inline inlong-sort/sort-connectors/hive/src/main/java/org/apache/inlong/sort/hive/filesystem/AbstractStreamingWriter.java:216]: dirty metric can't be reported if throw RumtimeException

Comment (haifxu) [reply inlong-sort/sort-connectors/hive/src/main/java/org/apache/inlong/sort/hive/filesystem/AbstractStreamingWriter.java]: Ok, don't throw exceptions temporarily.

File: inlong-sort/sort-connectors/hive/src/main/java/org/apache/inlong/sort/hive/filesystem/AbstractStreamingWriter.java [hunk 2]
```diff
@@ -195,14 +199,20 @@ public void processWatermark(Watermark mark) throws Exception {
...
+        try {
+            helper.onElement(
+                    element.getValue(),
+                    getProcessingTimeService().getCurrentProcessingTime(),
+                    element.hasTimestamp() ? element.getTimestamp() : null,
+                    currentWatermark);
+            if (metricData != null) {
+                metricData.invokeWithEstimate(element.getValue());
+            }
+        } catch (Exception e) {
```
Comment (healchow) [inline inlong-sort/sort-connectors/hive/src/main/java/org/apache/inlong/sort/hive/filesystem/AbstractStreamingWriter.java:212]: Is it necessary to print the exception info in the log?

Comment (haifxu) [reply inlong-sort/sort-connectors/hive/src/main/java/org/apache/inlong/sort/hive/filesystem/AbstractStreamingWriter.java:212]: I will improve in the next pr, it should support dirty data side-output.

File: inlong-sort/sort-connectors/hive/src/main/java/org/apache/inlong/sort/hive/filesystem/AbstractStreamingWriter.java [hunk 3]
```diff
@@ -195,14 +199,22 @@ public void processWatermark(Watermark mark) throws Exception {
...
     @Override
-    public void processElement(StreamRecord<IN> element) throws Exception {
-        helper.onElement(
-                element.getValue(),
-                getProcessingTimeService().getCurrentProcessingTime(),
-                element.hasTimestamp() ? element.getTimestamp() : null,
-                currentWatermark);
-        if (metricData != null) {
-            metricData.invokeWithEstimate(element.getValue());
+    public void processElement(StreamRecord<IN> element) {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/hive/src/main/java/org/apache/inlong/sort/hive/filesystem/AbstractStreamingWriter.java:202]: keep the method signature?

Comment (haifxu) [reply inlong-sort/sort-connectors/hive/src/main/java/org/apache/inlong/sort/hive/filesystem/AbstractStreamingWriter.java]: Done.

**Review by EMsnap — APPROVED** [review]

**Review by thesumery — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Pull request merged**